instrumentation/vaadin-14.2/testing/src/main/groovy/test/vaadin/AbstractVaadin14Test.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test.vaadin
+
+import static io.opentelemetry.instrumentation.test.utils.TraceUtils.basicSpan
+
+import com.vaadin.flow.server.Version
+
+abstract class AbstractVaadin14Test extends AbstractVaadinTest {
+  static final boolean VAADIN_14_4 = Version.majorVersion >= 2 && Version.minorVersion >= 4
+
+  List<String> getRequestHandlers() {
+    List<String> handlers = [
+      "PushRequestHandler"
+    ]
+    if (VAADIN_14_4) {
+      handlers.add("DevModeHandler")
+    }
+    handlers.addAll([
+      "StreamRequestHandler", "UnsupportedBrowserHandler", "UidlRequestHandler",
+      "HeartbeatHandler", "SessionRequestHandler", "FaviconHandler", "BootstrapHandler"
+    ])
+
+    return handlers
+  }
+
+  @Override
+  void assertFirstRequest() {
+    assertTraces(VAADIN_14_4 ? 5 : 4) {
+      def handlers = getRequestHandlers("BootstrapHandler")
+      trace(0, 3 + handlers.size()) {
+        basicSpan(it, 0, getContextPath() + "/main", null)
```

Comment:
I've just realized that `basicSpan()` does not check span kind at all...
`span(0)` should always be a SERVER span, right? Can you add a helper method in the base test class that does that? (I assume changing `basicSpan()` should be left for another PR)

Reply:
Added server span assertion.